checks.d/splunk_topology.py

```diff
@@ -0,0 +1,190 @@
+
+        if instance["url"] not in self.instance_data:
+            self.instance_data[instance["url"]] = Instance(instance, self.init_config)
+
+        instance = self.instance_data[instance["url"]]
+        current_time = self._current_time_seconds()
+        if not instance.should_poll(current_time):
+            return
+
+        instance_key = instance.instance_key
+
+        search_ids = [(self._dispatch_saved_search(instance.instance_config, saved_search), saved_search)
+                      for saved_search in instance.saved_searches]
+
+        self.start_snapshot(instance_key)
+
+        try:
+            for (sid, saved_search) in search_ids:
+                response = self._search(instance.instance_config, sid)
+                if saved_search.element_type == "component":
+                    self._extract_components(instance, response)
+                elif saved_search.element_type == "relation":
+                    self._extract_relations(instance, response)
+
+            # If everything was successful, update the timestamp
+            instance.last_successful_poll = current_time
+        finally:
+            self.stop_snapshot(instance_key)
+
+    def _current_time_seconds(self):
+        return int(round(time.time() * 1000))
+
+    def _search(self, instance_config, search_id):
+        """
+        perform a search operation on splunk given a search id (sid)
+        :param instance_config: current check configuration
+        :param search_id: perform a search operation on the search id
+        :return: raw response from splunk
+        """
+        search_url = '%s/services/search/jobs/%s/results?output_mode=json&count=0' % (instance_config.base_url, search_id)
+        auth = instance_config.get_auth_tuple()
+
+        response = requests.get(search_url, auth=auth, timeout=instance_config.timeout)
+        retry_count = 0
+
+        # retry until information is available.
+        while response.status_code == 204: # HTTP No Content response
+            if retry_count == instance_config.max_retry_count:
+                raise CheckException("maximum retries reached for " + instance_config.base_url + " with search id " + search_id)
+            retry_count += 1
+            time.sleep(instance_config.seconds_between_retries)
+            response = requests.get(search_url, auth=auth, timeout=instance_config.timeout)
+
+        return response.json()
+
+    def _dispatch_saved_search(self, instance_config, saved_search):
+        """
+        Initiate a saved search, returning the saved id
+        :param instance_config: Configuration of the splunk instance
+        :param saved_search: Configuration of the saved search
+        :return:
+        """
+        dispatch_url = '%s/services/saved/searches/%s/dispatch' % (instance_config.base_url, quote(saved_search.name))
+        auth = instance_config.get_auth_tuple()
+
+        parameters = saved_search.parameters[0]
```

Comment:
good point, thanks. figured out that the wrong yaml syntax was used:
```
parameters:
  - key: value
```
results in [{key:value}] while
```
parameters:
  key: value
```
results in {key:value}